<?xml version="1.0"?>
<robot name="RobotArm">
  <material name="blue">
    <color rgba="0 0 0.8 1" />
  </material>

  <material name="black">
    <color rgba="0 0 0 1" />
  </material>

  <material name="green">
    <color rgba="0 0.8 0 1" />
  </material>

  <link name="floor">
    <visual>
      <geometry>
        <box size="10 10 0.001" />
      </geometry>
      <origin xyz="0 0 -0.0005" />
      <material name="black" />
    </visual>
  </link>
  
  <link name="link_0">
    <visual>
      <geometry>
        <cylinder length="0.09" radius="0.06" />
      </geometry>
      <origin xyz="0 0 0.045" />
    </visual>
  </link>
  
  <joint name="floor_to_link_0" type="revolute">
    <axis xyz="0 0 1" />
    <limit effort="100.0" lower="-1.57075" upper="1.57075" velocity="0.5" />
    <origin rpy="0 0 0" xyz="0 0 0" />
    <parent link="floor" />
    <child link="link_0" />
  </joint>

  
  <link name="link_1">
    <visual>
      <geometry>
        <box size="0.055 0.025 0.104" />
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.052" />
      <material name="blue"/>
    </visual>
  </link>

  <joint name="link_0_to_link_1" type="revolute">
    <axis xyz="1 0 0" />
    <limit effort="100.0" lower="-1.57075" upper="1.57075" velocity="0.5" />
    <origin rpy="0 0 0" xyz="0 0.01 0.09" />
    <parent link="link_0" />
    <child link="link_1" />
  </joint>

  
  <link name="link_2">
    <visual>
      <geometry>
        <box size="0.055 0.025 0.089" />
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.0445" />
      <material name="green" />
    </visual>
  </link>

  <joint name="link_1_to_link_2" type="revolute">
    <axis xyz="1 0 0" />
    <limit effort="100.0" lower="-1.57075" upper="1.57075" velocity="0.5" />
    <origin rpy="0 0 0" xyz="0 0 0.104" />
    <parent link="link_1" />
    <child link="link_2" />
  </joint>


  <link name="link_3">
    <visual>
      <geometry>
        <box size="0.025 0.055 0.057" />
      </geometry>
      <origin rpy="0 0 0" xyz="0 0.015 0.0285" />
      <material name="green" />
    </visual>
  </link>

  <joint name="link_2_to_link_3" type="revolute">
    <axis xyz="1 0 0" />
    <limit effort="100.0" lower="-1.57075" upper="1.57075" velocity="0.5" />
    <origin rpy="0 0 0" xyz="0 0 0.089" />
    <parent link="link_2" />
    <child link="link_3" />
  </joint>

  <link name="gripper_pole">
    <visual>
      <geometry>
        <cylinder length="0.02" radius="0.01" />
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.01" />
      <material name="black" />
    </visual>
  </link>

  <joint name="link_3_to_gripper_pole" type="revolute">
    <axis xyz="0 0 1" />
    <limit effort="100.0" lower="-1.57075" upper="1.57075" velocity="0.5" />
    <origin rpy="0 0 0" xyz="0 0 0.057" />
    <parent link="link_3" />
    <child link="gripper_pole" />
  </joint>

  <!--Tool Frame-->
  <joint name="left_gripper_joint" type="revolute">
    <axis xyz="1 0 0" />
    <limit effort="1000.0" lower="-0.548" upper="0.0" velocity="0.5" />
    <origin rpy="0 0 0" xyz="0 0.01 0.02" />
    <parent link="gripper_pole" />
    <child link="left_gripper" />
  </joint>

  <link name="left_gripper">
    <visual>
      <origin rpy="0.0 -1.57075 0" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://urdf_tutorial/meshes/l_finger.dae" />
      </geometry>
    </visual>
  </link>

  <joint name="left_tip_joint" type="fixed">
    <parent link="left_gripper" />
    <child link="left_tip" />
  </joint>

  <link name="left_tip">
    <visual>
      <origin rpy="0.0 -1.57075 0" xyz="0 0.00495 0.09137" />
      <geometry>
        <mesh filename="package://urdf_tutorial/meshes/l_finger_tip.dae" />
      </geometry>
    </visual>
  </link>

  <joint name="right_gripper_joint" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="0.0" upper="0.548" velocity="0.5" />
    <origin rpy="0 0 0" xyz="0 -0.01 0.02" />
    <parent link="gripper_pole" />
    <child link="right_gripper" />
  </joint>

  <link name="right_gripper">
    <visual>
      <origin rpy="-3.1415 -1.57075 0" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://urdf_tutorial/meshes/l_finger.dae" />
      </geometry>
    </visual>
  </link>

  <joint name="right_tip_joint" type="fixed">
    <parent link="right_gripper"/>
    <child link="right_tip"/>
  </joint>

  <link name="right_tip">
    <visual>
      <origin rpy="-3.1415 -1.57075 0" xyz="0 -0.00495 0.09137"/>
      <geometry>
        <mesh filename="package://urdf_tutorial/meshes/l_finger_tip.dae"/>
      </geometry>
    </visual>
  </link>

 

</robot>
